### ISSUE TITLE
Grid not followed sometimes after section

### ISSUE THREAD

**Issue by jslabovitz** [sile-typesetter/sile#1720]
In the book I'm typesetting, I'm using grid mode. (Frankly, it's one of the main reasons I'm using SILE.) Unfortunately I'm frequently seeing situations where the grid is not being following. Commonly, this involves the text lines that occur after sections. Here's an example, though there seem to be some other pathological cases where it can occur also. (See page 4 for the text that is not aligned to the grid.)

Is there something else I need to set in order to make grids work more consistently? 

```
\begin[class=book,papersize="a5"]{document}
\use[module="packages.grid"]
\grid[spacing="18pt"]
\grid:debug
% grid fails regardless whether following settings are active or not
\font[size="14pt"]
\set[parameter="document.parskip", value="0"]
\set[parameter="document.baselineskip", value="18pt"]
\set[parameter="document.lineskip", value="0"]
%
\chapter{My chapter}

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

sile

\section{My section}

sile

sile

sile

\end{document}
```

**Comment by jslabovitz** [comment]
I'm sorry to take folks' time in raising this issue again. But I'm finding that the `grid` package is really not working properly. In almost every document I try to build, lines are frequently not aligning as expected — particularly after chapter or section headings.

Is anyone else actually using this package?

**Comment by Omikhleia** [comment]
Isn't it our good old obscure friend `SILE.typesetter:inhibitLeading()` which breaks the grid leading. See #1300, #1288 -- That you actually annotated a month ago?

**Comment by jslabovitz** [comment]
Yes, apologies for the lapse of memory. I guess I keep seeing this and thinking it’s another bug, but perhaps it’s all the same thing.

I patched my local installation to disable the action of inhibitLeading() (by just commented out the code), and that does seem to fix most of the grid problems.

However, I’m still seeing grid misalignments in \tableofcontents. But at this point I’m giving up.

**Comment by Omikhleia** [comment]
> But at this point I’m giving up.

Oh, I know that feeling :)

Beyond the issue/question regarding the sections inhibiting leaders in the standard book class, the grid package is probably not used seriously, it's easy to break (even without sections and ToCs). It's intended for some bible typesetting, but I am not aware of any serious public bible project... Unless your project is indeed a bible? Otherwise, I'd question the real need for grid typesetting (esp. with the default ToC level commands, which have plenty of skips and font size change, likely to look ugly even if the grid worked).

**Comment by jslabovitz** [comment]
My project is not a bible. It’s a friend’s personal memoir, of about 500pp. I consider it a fairly simple & straightforward design: front matter with title page, copyright, prologue, TOC; main matter with two parts and about 60 short chapters; and back matter with a glossary. It has no running heads or footnotes, and only simple folios. It’s all ragged-right, with light hyphenation, mostly 14/18 text.

Really, it’s just a typical sort of humanistic, non-technical book. It surprises me that for such a simple work, the kind I’ve designed many times over my 40 years of being a typographer, it’s been such a chore to make things work in SILE. I know, it’s free and open source, but still, hearing that grids perhaps *shouldn’t* be used, even though that’s a ‘front page' feature, is a bit frustrating.

Maybe I’ve been making bad assumptions, but controllable and consistent linespacing using grids, or leading and space before/after make-up, is a basic thing for my work, akin to the ability to specify wordspacing, kerning, or letterspacing. I guess it’s a philosophical difference, grids being on one side, and things like `smallskip` and flexible vertical glue on the other. I’m not sure how to resolve that, without implementing my own classes from scratch.

**Comment by ctrlcctrlv** [comment]
I totally understand your frustration and perhaps we should update the website as the things Simon was focusing on when he wrote SILE aren't really the project's main foci anymore.

For example, Simon was a missionary (I think?? Maybe misremembering) and is a devout Christian, so that's where grid comes from. I've never used it (I'm also a Christian just far less devout lol), and I _don't_ think it should be front page advertised if it has serious bugs as @Omikhleia said.

Maybe, @Omikhleia, since you're one of the most active contributors, you can help perhaps Caleb if he wants to and me figure out what we should be telling people SILE can do, given the parts Simon cared the most about but few of us do have kinda fallen by the wayside?

**Comment by ctrlcctrlv** [comment]
> **IMHO, SILE brags a lot (and far too much)**... 

Agreed, @Omikhleia.

Hadn't thought about it since this was posted as the first and last time I first read the website (until today) was years ago. We can move this into a discussion though so as not to clutter this issue with project meta: #1729. :-)

**Comment by Omikhleia** [comment]
> but they're floating a few points below the baseline.

Trying to think what else can cause it... Another point comes to my mind, perhaps unrelated, but the first line on a page doesn't always start at the same place, esp. if it doesn't have ascenders: #1430 (the main issue was fix for the grid, not fixed for the TeX linespacing as being consistent with what the latter does; I am unsure the last (grid-related) point was really addressed...)

**Comment by CatholicPenguin** [comment]
Just a note about the purpose of SILE: I have been following SILE for some time because I do almost exclusively religious typesetting, like Bibles and Missals. Should I move to another platform? LaTeX isn't the best, but will I have a better time there in five years, for example?
